cli/update.go

```diff
@@ -23,18 +30,38 @@ func update() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			if !workspace.Outdated {
+			if !workspace.Outdated && !alwaysPrompt {
 				_, _ = fmt.Printf("Workspace isn't outdated!\n")
 				return nil
 			}
 			template, err := client.Template(cmd.Context(), workspace.TemplateID)
 			if err != nil {
 				return nil
 			}
+
+			var existingParams []codersdk.Parameter
+			if !alwaysPrompt {
+				existingParams, err = client.Parameters(cmd.Context(), codersdk.ParameterWorkspace, workspace.ID)
+				if err != nil {
+					return nil
+				}
+			}
+
+			parameters, err := prepWorkspaceBuild(cmd, client, prepWorkspaceBuildArgs{
+				Template:         template,
+				ExistingParams:   existingParams,
+				ParameterFile:    parameterFile,
+				NewWorkspaceName: workspace.Name,
+			})
```

Comment:
It would be nice if this always ran and had an argument to prompt for missing params or not instead of only running if they prompted for params. We'd be able to always dry-run this way

Reply:
You mean instead of passing "ExistingParams", detect that inside the `prepWorkspaceBuild`? Cause if so, I do want to do this. Templates use similar logic, so I want to refactor the create/update of both workspaces/templates to use the same code.

I was going to take a look in another PR.